api/api.go

```diff
@@ -117,12 +118,13 @@ func getContainersJSON(c *context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	all := r.Form.Get("all") == "1"
+	limit, _ := strconv.Atoi(r.Form.Get("limit"))
 
 	out := []*dockerclient.Container{}
 	for _, container := range c.cluster.Containers() {
 		tmp := (*container).Container
 		// Skip stopped containers unless -a was specified.
-		if !strings.Contains(tmp.Status, "Up") && !all {
+		if !strings.Contains(tmp.Status, "Up") && !all && limit <= 0 {
```

Comment:
nit: Maybe the limit could be checked outside the for loop since it doesn't make sense to iterate over all containers if limit is 0.

Then again, I think it would be a useless optimization: limit of 0 is a strange limit and it doesn't make much sense to optimize for this.

In short: up to you :) feel free to discard this